flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -277,6 +277,7 @@ public CheckpointCoordinator(
 
 		try {
 			this.checkpointStorage = checkpointStateBackend.createCheckpointStorage(job);
+			checkpointStorage.setCheckpointProperties(checkpointProperties);
```

Comment:
`CheckpointStorage` is used on both JM and TM side, and the `checkpointProperties` is only required by `CheckpointCoordinator`, so it should not be a property in `CheckpointStorage`. Instead, we should pass it in as a parameter in `CheckpointStorageCoordinatorView#shutDown`.

Accordingly, we don't need to introduce the setter interface in `CheckpointStorageCoordinatorView` any more.